README.md

```diff
@@ -30,13 +35,67 @@ The basename of each `*.groovy` file should be a Groovy (~ Java) identifier, con
 The matching `*.txt`, if present, can contain documentation, processed through the system’s configured markup formatter
 (so may really be HTML, Markdown, etc., though the `txt` extension is required).
 
-The groovy source files in these directories get the same sandbox / CPS
-transformation just like your Pipeline scripts.
+The Groovy source files in these directories get the same “CPS transformation” as your Pipeline scripts.
+
+A `resources` directory allows the `libraryResource` step to be used from an external library to load associated non-Groovy files.
+Currently this feature is not supported for internal libraries.
 
 Other directories under the root are reserved for future enhancements.
 
+## Defining external libraries
+
+An external library is defined with a name, an SCM, and optionally a default version.
+The name should be a short identifier as it will be used in scripts.
+
+The version could be anything understood by that SCM; for example, branches, tags, and commit hashes all work for Git.
+You may also declare whether scripts need to explicitly request that library (detailed below), or if it is present by default.
+Furthermore, if you specify a version in Jenkins configuration, you can block scripts from selecting a _different_ version.
+
+The best way to specify the SCM is using an SCM plugin which has been specifically updated
+to support a new API for checking out an arbitrary named version.
+Initially the Git plugin has this modification; others should follow.
+
+If your SCM plugin has not been integrated, you may select _Single repository & branch_ and pick anything offered.
+In this case, you need to include `${library.yourLibName.version}` somewhere in the configuration of the SCM,
+so that during checkout the plugin will expand this variable to select the desired version.
+For example, for Subversion, you can set the _Repository URL_ to `https://svnserver/project/${library.yourLibName.version}`
+and then use versions such as `trunk` or `branches/dev` or `tags/1.0`.
+
+### Global external libraries
+
+There are several places you can define external libraries, according to your needs.
+
+Under _Manage Jenkins » Configure System » Global Pipeline Libraries_ you may configure as many libraries as you need.
+These will be accessible to any Pipeline job in the system.
+
+These libraries are trusted: they can run any methods in Java, Groovy, Jenkins internal APIs, Jenkins plugins, or third-party libraries.
```

Comment:
Well it is just up to the Jenkins administrator to implement proper security controls on that repository. Using nondeterministic (e.g., branch) revisions on global libraries is an important use case.

The administrator already has the option of defining the library with a nonoverridable version pointing to a tag or (better) commit hash.